Integrating Husky Pre-Commit Hook with ESLint and Prettier

Naman-B-Parlecha opened the pull request:
## Description  
This PR introduces a **Husky pre-commit hook** to automatically run **ESLint** and **Prettier** before committing code.  
It ensures that only properly linted and formatted code is committed, preventing errors and enforcing consistency across the codebase.  

## Why is this needed?  
- Prevents committing code with linting errors.  
- Ensures consistent formatting, reducing unnecessary PR changes.  
- Saves time in code reviews and CI by catching issues early.  

## Changes Introduced 
 
- [x]  Added **Husky** pre-commit hook.  
- [x] Integrated **ESLint** for linting staged files.  
- [x] Configured **Prettier** for automatic formatting.  
- [x] Ensured the hook runs only on staged files before commit.  

## steps to replicate and test the pr

```bash
# Navigate to the frontend directory
cd frontend

# Install dependencies
npm install

# Make changes to any file and stage them
git add <file-name>

# Try committing the changes
git commit -m "Test commit"
```

## Type
- [x] New Feature
- [ ] Bug Fix
- [ ] Refactoring
- [ ] Documentation

### Commiting with no Error

![Screenshot from 2025-02-13 00-31-33](https://github.com/user-attachments/assets/24e11d4d-9444-418a-8a80-071bcaf37cf9)

### Commiting with Errors
![Screenshot from 2025-02-13 00-31-28](https://github.com/user-attachments/assets/5ec08154-bee5-45ed-8a2d-f84c0ef36989)

Naman-B-Parlecha commented:
@keshav-nischal can u review this!

keshav-nischal commented on frontend/package.json:
```diff
@@ -16,6 +20,7 @@
     "@radix-ui/react-scroll-area": "^1.2.2",
     "@radix-ui/react-separator": "^1.1.0",
     "@radix-ui/react-slot": "^1.1.0",
+    "aws-amplify": "^6.13.0",
```
check if we are using this somewhere and remove this if not required

Naman-B-Parlecha replied:
this is not being used anywhere 
hence removing this dependency

keshav-nischal commented on frontend/src/Pages/Game.tsx:
```diff
@@ -108,9 +112,8 @@ const Game: React.FC = () => {
 
     return () => ws.close();
-  }, [userId]);
+  }, [userId, handleWebSocketMessage]);
```
this change doesn't seem related to PR, can you check why this is required

Naman-B-Parlecha replied:
this function is being used in the useEffect and a dependency of the function is needed to be added to it

i have refactored the handleWebSocketMessage function too, wrapping it with useCallback

keshav-nischal commented on frontend/src/Pages/Game.tsx:
```diff
@@ -162,10 +165,10 @@ const Game: React.FC = () => {
             micOn={state.micOn}
             setCameraOn={(value) =>
-              setState((prev) => ({ ...prev, cameraOn: value }))
+              setState((prev: any) => ({ ...prev, cameraOn: value }))
```
give appropriate type to this

Naman-B-Parlecha replied:
refactoring the values to boolean only

keshav-nischal commented on frontend/src/Pages/Game.tsx:
```diff
@@ -96,7 +101,6 @@ const Game: React.FC = () => {
   };
 
   useEffect(() => {
-    const userDetails = `${import.meta.env.VITE_BASE_URL/getUserDetail}`;
```
make sure this is not being used anywhere else

Naman-B-Parlecha replied:
yep checked it, i had removed it as it wasn't being used anywhere

Naman-B-Parlecha commented:
@keshav-nischal refactored as asked

keshav-nischal commented:
is this tested(the change is more than eslint, something related to dependency was also change)? for frontend
please mension the test scenarios over here

Naman-B-Parlecha commented:
Yep this is tested, I have added the screenshot in the description of this pr.

There was aws-amplify which was not being used anywhere so I removed it, apart from that I have added the recommended dependencies for prettier husky and eslints.
